test/functional/api/buffer_spec.lua

```diff
@@ -64,6 +67,28 @@ describe('api/buf', function()
       curbuf_depr('set_line', 0, 'ab\0cd')
       eq('ab\0cd', curbuf_depr('get_line', 0))
     end)
+
+    it("set_line on alternate buffer doesn't give Vim error E315", function()
+      execute('set hidden')
+      helpers.source([[
+      function! SetLines()
+        call nvim_buf_set_lines(1, 0, 1, v:false, ['test'])
+      endfunction
```

Comment:
Let's move this test to the 

     describe('{get,set}_lines', function()

group. Currently it's in the "deprecated" section (old API)